builtin/credential/app-id/backend_test.go

```diff
@@ -93,8 +93,9 @@ func TestBackend_displayName(t *testing.T) {
 
 func testAccStepMapAppId(t *testing.T) logicaltest.TestStep {
 	return logicaltest.TestStep{
-		Operation: logical.UpdateOperation,
-		Path:      "map/app-id/foo",
+		IsAuthBackendRequest: true,
```

Comment:
Oooh, good idea, I'll give that a go.

Reply:
So I tried that and it turns out that it fails on the simplest implementation. The reason is, the mount _also_ gets prepended to the path. So, as written, this test step ultimately produces this request path:
`auth/mnt/map/app-id/foo`

If I strip the logic around `IsAuthBackendRequest`, and prepend the path with `auth/`, it ultimately produces this path: 
`mnt/auth/map/app-id/foo`

That fails because of auth and mnt being switched.

So, one solution for that might be that `IsAuthBackendRequest` could be stripped, but there could be a template field for the mount in the path? Like: `Path: "auth/{{mount}}/map/app-id/foo"`. Let me know your opinion.

Reply:
I see. If we are always using a single backed for these tests, can we derive whether we need to hit a `auth/` endpoint within `func Test` itself, based on whether the backend is of type `LogicalBackend` or `CredentialBackend`. We are already checking in here: https://github.com/hashicorp/vault/pull/5536/files/e8f495468e223439f7f59ea477427cc9a8acb150#diff-595770755da51da402d24c346acebf72R250?

Reply:
Ah yes! That should work. Will do.

Reply:
This is pushed.